fix(TransitiveOverlay): Expose styles and labeledModes transitive props.

binh-dam-ibigroup opened the pull request:
For some reason I was convinced the transitive options were directly exposed but I was wrong... so we have to expose the new styles and labeledModes on the components.

EDIT: In response to https://github.com/opentripplanner/otp-ui/pull/246#discussion_r630499498, this PR now requires #248. To test, merge the changes from #248 into this branch and run `yarn dev`.

binh-dam-ibigroup commented:
Sorry, the PR is not working as expected, I'm reverting to draft and assigning to myself.

binh-dam-ibigroup commented:
Ready for review again, thanks for your patience

fpurcell commented on packages/transitive-overlay/src/TransitiveOverlay.story.js:
```diff
@@ -26,6 +31,11 @@ const walkTransitWalkItinerary = require("@opentripplanner/itinerary-body/src/__
+// Add routeShortName to walk-transit-walk itineary here
+// to illustrate labeledModes in the transitive overlay
+// (and to not interfere with other stories that use that itinerary).
+walkTransitWalkItinerary.legs[1].routeShortName = "Blue";
```
Off the top of my head, I worry about how controlling the label content via routeShortName will work in the real world, Bihn.

 - hoping there's another way to change the content of the route label content in OTP-UI + transitive-js beyond monkey-patching routeShortName 
 - was imagining my client would provide OTP-UI an optional slot or callback, which transitive-js call whenever  rendering a route label:
   function renderRouteLabelSlot(itineraryLeg) {
       if (itineraryLeg.mode === "TRAM") return "MAX";
       if (itineraryLeg.mode === "RAIL") return "WES";
       if (itineraryLeg.mode === "BUS") return itineraryLeg.routeShortName; 
       return null;  // null or undefined or empty string will tell transitive-js not to render a route label
  }

binh-dam-ibigroup replied:
@fpurcell and all, please see #248 as an option to accommodate the custom route label function you suggested above. I guess there is no need to change anything in transitive.js at this time with this approach.

binh-dam-ibigroup commented:
Marking this PR as blocked pending reviews for #248.

fpurcell reviewed (approved):
Think this will work well -- from what I gather, the Storybook example will work once #248 is merged, and this PR re-based (e.g., the custom SB example is not yet showing a route label yet, but anticipate 'MAX' showing up once all is merged / based).

evansiroky reviewed (changes requested):
So far so good, but assigning back to Binh to bump to the latest version of core-utils in the transitive-overlay package.

binh-dam-ibigroup commented:
Ready for review again. Excluding the changes from `master` from https://github.com/opentripplanner/otp-ui/pull/246/commits/d231abf9e9cd1b9b637892941d4b1a6ec87e5b45 and the bump in version numbers there are no other code changes since last submission.

fpurcell reviewed (approved):
Working for me ... ship it!
